src/main/resources/assets/js/components/Marathon.jsx

```diff
@@ -43,11 +49,61 @@ define([
         deploymentsFetchState: States.STATE_LOADING,
         fetchState: States.STATE_LOADING,
         modalClass: null,
+        route: null,
         tasksFetchState: States.STATE_LOADING
       };
     },
 
     componentDidMount: function() {
+      var router = this.props.router;
+
+      function callRoute(routeFunc) {
```

Comment:
Can you explain this? I don't want to expose this recursive helper function to the whole class and I want to hold it "private".